### PR TITLE
457 enhance handlers to support attach and logger

### LINKED ISSUE #457: Enhance handlers to support attach and logger
**Is your feature request related to a problem? Please describe.**
Currently, the TensorBoard image handler is very simple, need to support `attach` API.
Not all the handlers support `logging`, need to unify that.
And need to add `FileHandler` to StatsHandler as Eric suggested.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #457 .

### Description
This PR implemented below items:
Added TensorBoard image handler support for `attach` API.
And added FileHandler to StatsHandler as Eric suggested.
Refined several functions.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: monai/handlers/tensorboard_handlers.py [hunk 1]
```diff
@@ -194,6 +196,9 @@ def __init__(
             summary_writer (SummaryWriter): user can specify TensorBoard SummaryWriter,
                 default to create a new writer.
             log_dir (str): if using default SummaryWriter, write logs to this directory, default is `./runs`.
+            interval (int): plot image every N epochs or every N iterations, default is 1.
```
Comment (wyli) [inline monai/handlers/tensorboard_handlers.py:199]: plot image -> plot content from `engine.state`

File: monai/handlers/tensorboard_handlers.py [hunk 2]
```diff
@@ -194,6 +196,9 @@ def __init__(
...
+            epoch_level (bool): plot image every N epochs or N iterations. `True` is epoch level,
```
Comment (wyli) [inline monai/handlers/tensorboard_handlers.py:200]: plot image -> plot content from `engine.state`

File: monai/handlers/tensorboard_handlers.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handlers.py]: updated in the latest patch.
Thanks.

File: monai/handlers/tensorboard_handlers.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/handlers/tensorboard_handlers.py]: updated in the latest patch.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your coments.
I updated the PR.

**Comment by Nic-Ma** [comment]
Hi @ericspod , this PR already added support to `FileHandler` to StatsHandler as you suggested.
Thanks.

**Pull request merged**